src/feature.js

```diff
@@ -144,12 +100,32 @@ class FxABrowserFeature {
 
       ctx.drawImage(this, 0, 0);
 
-      this._avatar = ctx.getImageData(0, 0, 200, 200);
+      const imageData = ctx.getImageData(0, 0, 200, 200);
 
-      browser.browserAction.setIcon({ imageData: this._avatar });
-    }
+      browser.browserAction.setIcon({ imageData });
+    };
     img.src = url;
   }
+
+  sendTelemetry(data) {
+    browser.study.sendTelemetry({
+      ...data,
+      addonId: "fxadisco",
+      addonVersion: "1",
+      branch: this._variation,
+      startTime: `${Date.now()}`,
+      fxaState: this._state || "none",
+      hasAvatar: `${!!this._avatar}`,
```

Comment:
The reason I wanted to call this `sendTelemetry` method from the popup was so that we'd have access to all this cached state for the boilerplate properties.